marathon/client.py

```diff
@@ -167,16 +178,29 @@ def list_apps(self, cmd=None, embed_tasks=False,
             apps = [o for o in apps if getattr(o, k) == v]
         return apps
 
-    def get_app(self, app_id, embed_tasks=False):
+    def get_app(self, app_id, embed_tasks=False, embed_counts=False,
+                embed_deployments=False, embed_readiness=False,
+                embed_last_task_failure=False, embed_failures=False,
+                embed_task_stats=False):
         """Get a single app.
 
         :param str app_id: application ID
         :param bool embed_tasks: embed tasks in result
+        :param bool embed_counts: embed all task counts
+        :param bool embed_deployments: embed all deployment identifier
+        :param bool embed_readiness: embed all readiness check results
+        :param bool embed_last_task_failure: embeds the last task failure
+        :param bool embed_failures: shorthand for embed_last_task_failure
+        :param bool embed_task_stats: embed task stats in result
 
         :returns: application
         :rtype: :class:`marathon.models.app.MarathonApp`
         """
-        params = {'embed': 'apps.tasks'} if embed_tasks else {}
+        params = {}
+        embed_param = self._parse_embed_params()
```

Comment:
Agreed that inspect is magical to achieve less redundant code, otherwise these params code should appear twice in get_app and list_apps. If these two methods are not merged into one, it seems hard to avoid redundancy. If the redundancy is acceptable, I am ok to change the code in this way.